keyboards/macroflow_original/keymaps/default/keymap.c

```diff
@@ -0,0 +1,297 @@
+// Copyright 2024 Patrick Mathern (@Patrickemm)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+// Define the keycode, `QK_USER` avoids collisions with existing keycodes
+enum keycodes {
+  KC_CYCLE_LAYERS = QK_USER,
+};
+
+enum layer_names {
+  one,
+  two,
+  three,
+  four,
+  five,
+  six,
+  seven,
+  eight
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [0] = LAYOUT_macroflow_2x4(
+        KC_0, KC_CYCLE_LAYERS,
+        KC_1, KC_2, KC_3, KC_4,
+        KC_5, KC_6, KC_7, KC_8
+    ),
+    [1] = LAYOUT_macroflow_2x4(
+        KC_9, KC_CYCLE_LAYERS,
+        KC_A, KC_B, KC_C, KC_D,
+        KC_E, KC_F, KC_G, KC_H
+    ),
+    [2] = LAYOUT_macroflow_2x4(
+        KC_I, KC_CYCLE_LAYERS,
+        KC_J, KC_K, KC_L, KC_M,
+        KC_N, KC_O, KC_P, KC_Q
+    ),
+    [3] = LAYOUT_macroflow_2x4(
+        KC_R, KC_CYCLE_LAYERS,
+        KC_S, KC_T, KC_U, KC_V,
+        KC_W, KC_X, KC_Y, KC_Z
+    ),
+    [4] = LAYOUT_macroflow_2x4(
+        KC_TRNS, KC_CYCLE_LAYERS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [5] = LAYOUT_macroflow_2x4(
+        KC_TRNS, KC_CYCLE_LAYERS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [6] = LAYOUT_macroflow_2x4(
+        KC_TRNS, KC_CYCLE_LAYERS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [7] = LAYOUT_macroflow_2x4(
+        KC_TRNS, KC_CYCLE_LAYERS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
+};
+
+// 1st layer on the cycle
+#define LAYER_CYCLE_START 0
+// Last layer on the cycle
+#define LAYER_CYCLE_END 8
```

Comment:
You could actually check the array size. And `keymap_layer_count()` may work for this, specifically (and be the best).